cli/formatter/table.go

```diff
@@ -0,0 +1,492 @@
+	}
+
+	if opts.TableDialect == MarkdownTableDialect {
+		markdown := strings.Split(t.RenderMarkdown(), "\n")
+		return createMarkdownTable(markdown, columnsAmount, opts) + "\n", nil
+	}
+	if opts.TableDialect == JiraTableDialect {
+		return t.RenderMarkdown() + "\n\n", nil
+	}
+
+	return t.Render() + "\n", nil
+}
+
+// isMapKeysEqual checks keys equal for maps[string]any.
+func isMapKeysEqual(x map[string]any, y map[string]any) bool {
+	var keysX, keysY []string
+
+	for k := range x {
+		keysX = append(keysX, k)
+	}
+	for k := range y {
+		keysY = append(keysY, k)
+	}
+
+	if len(keysX) != len(keysY) {
+		return false
+	}
+
+	sort.Strings(keysX)
+	sort.Strings(keysY)
+	for k, v := range keysX {
+		if keysY[k] != v {
+			return false
+		}
+	}
+
+	return true
+}
+
+// renderBatch parses renders batch and return tables as string for it.
+func renderBatch(batch []any, transpose bool, opts Opts) (string, error) {
+	if isSingleType(batch, scalarNodeType) {
+		return renderScalars(batch, transpose, opts)
+	} else if isSingleType(batch, mapNodeType) {
+		var anyMaps []map[string]any
+		for _, node := range batch {
+			castedMap := castAnyMapToStringMap(node.(map[any]any))
+			anyMaps = append(anyMaps, castedMap)
+		}
+
+		var mapsBatchs = make([][]map[string]any, len(anyMaps))
+		var batchPointer = 0
+		mapsBatchs[batchPointer] = append(mapsBatchs[batchPointer], anyMaps[0])
+
+		for i := 0; i < len(anyMaps)-1; i++ {
+			if !isMapKeysEqual(anyMaps[i], anyMaps[i+1]) {
+				batchPointer = batchPointer + 1
+			}
+			mapsBatchs[batchPointer] = append(mapsBatchs[batchPointer], anyMaps[i+1])
+		}
+
+		var res, batchRes string
+		var err error
+		for _, batch := range mapsBatchs {
+			if len(batch) != 0 {
+				batchRes, err = renderEqualMaps(batch, transpose, opts)
+				if err != nil {
+					return "", err
+				}
+				if !opts.Graphics {
+					batchRes = batchRes + "\n"
+				}
+				res = res + batchRes
+			}
+		}
+
+		return res, nil
+	} else if isSingleType(batch, arrayNodeType) {
+		return renderArrays(batch, transpose, opts)
+	} else {
+		return "", fmt.Errorf("unknown parsing case with current render batch")
+	}
+}
+
+// renderBatches combines multiple batches into one string.
+func renderBatches(batches [][]any, transpose bool, opts Opts) (string, error) {
+	var result string
+	for _, batch := range batches {
+		if len(batch) != 0 {
+			batchStr, err := renderBatch(batch, transpose, opts)
+			if err != nil {
+				return "", fmt.Errorf("cannot render tables: %w", err)
+			}
+			result += batchStr
+			if !opts.Graphics {
+				result += "\n"
+			}
+		}
+	}
+
+	return result, nil
+
+}
+
+type metadataField struct {
+	Name string `yaml:"name"`
+}
+
+type metadataRows struct {
+	Metadata []metadataField `yaml:"metadata"`
+	Rows     [][]any         `yaml:"rows"`
+}
+
+// remapMetadataRows creates maps from rows with a metainformation.
+func remapMetadataRows(meta metadataRows) []any {
+	var nodes []any
+	for _, row := range meta.Rows {
+		index := 1
```

Comment:
Is there any specific reason for `index` to start from 1 and not 0?

Reply:
Yes it is, Lua-tables with "array" parts are marshal-unmarshal with `1` as a key for the first index. So, Lua table `{foo = "bar", [1] = 1, [2] = 2}`  would be after unmarshaling:

`Go
map[any]any{
    "foo": "bar",
    1: "1",
    2: "2",
}
```

The code do the same mapping.